m4/ax_check_compile_flag.m4

```diff
@@ -0,0 +1,74 @@
+# ===========================================================================
+#  https://www.gnu.org/software/autoconf-archive/ax_check_compile_flag.html
+# ===========================================================================
+#
+# SYNOPSIS
+#
+#   AX_CHECK_COMPILE_FLAG(FLAG, [ACTION-SUCCESS], [ACTION-FAILURE], [EXTRA-FLAGS], [INPUT])
+#
+# DESCRIPTION
+#
+#   Check whether the given FLAG works with the current language's compiler
+#   or gives an error.  (Warnings, however, are ignored)
+#
+#   ACTION-SUCCESS/ACTION-FAILURE are shell commands to execute on
+#   success/failure.
+#
+#   If EXTRA-FLAGS is defined, it is added to the current language's default
+#   flags (e.g. CFLAGS) when the check is done.  The check is thus made with
+#   the flags: "CFLAGS EXTRA-FLAGS FLAG".  This can for example be used to
+#   force the compiler to issue an error when a bad flag is given.
+#
+#   INPUT gives an alternative input source to AC_COMPILE_IFELSE.
+#
+#   NOTE: Implementation based on AX_CFLAGS_GCC_OPTION. Please keep this
+#   macro in sync with AX_CHECK_{PREPROC,LINK}_FLAG.
+#
+# LICENSE
+#
+#   Copyright (c) 2008 Guido U. Draheim <guidod@gmx.de>
+#   Copyright (c) 2011 Maarten Bosmans <mkbosmans@gmail.com>
+#
+#   This program is free software: you can redistribute it and/or modify it
+#   under the terms of the GNU General Public License as published by the
+#   Free Software Foundation, either version 3 of the License, or (at your
```

Comment:
@tpetazzoni the Jose project is licensed under Apache 2.0 but this macro is licensed under GPLv3. So I wonder if it can be included in the project? My understanding is that Apache 2.0 code can be included in a GPLv3 project but not the opposite.

Reading the license section in https://www.gnu.org/software/autoconf-archive/ax_check_compile_flag.html, I see that there's a license exception for the output of the macro that ends in the configure script, but the macro itself is GPLv3.

License compliance is always a confusing topic for me and I don't know if autoconf macros are a special case, so please let me know if I'm wrong on this.

Reply:
@martinezjavier That is a very good question, and I must say I don't have the answer :-/

Reply:
@martinezjavier It depends on what the meaning of `derivative work` is. This is widely understood to mean linking. Basically, the build system of jose is under the GPLv3 license. But the code itself is ASL2. In the near future, I'd like to move to Meson anyway; so this is likely to become moot pretty soon.